app/src/androidTest/java/com/example/roome/EditProfileWoutPictureTest.java

```diff
@@ -0,0 +1,410 @@
+package com.example.roome;
+
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewParent;
+
+import androidx.test.espresso.ViewInteraction;
+import androidx.test.filters.LargeTest;
+import androidx.test.rule.ActivityTestRule;
+import androidx.test.runner.AndroidJUnit4;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.replaceText;
+import static androidx.test.espresso.action.ViewActions.scrollTo;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isChecked;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.isNotChecked;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+
+@LargeTest
+public class EditProfileWoutPictureTest {
+    /**
+     * This test checks that a user (searching for a room) updates his/her profile successfully
+     * (without picture upload).
+     * Actions:
+     *          - (Rule) start MainActivity
+     *          - enter signInFirstName as a first name, signInLastName as last name
+     *          - pick searching a room to join
+     *          - go to edit profile page, update the profile with info as given below
+     *          - press save button
+     *          - go to matches page, then come back to edit profile page
+     *          - check that all the previously entered info is displayed accordingly
+     *              =   end   =
+     */
+
+    // Sign in with:
+    private String signInFirstName  = "fn";
+    private String signInLastName   = "ln";
+
+    // Update profile with:
+    private String updateFirstName      = "Spongebob";
+    private String updateLastName       = "Squarepants";
+    private String updateAge            = "66";
+    private String updateGender         = "Female";
+    private String updatePhoneNumber    = "0541234567";
+    private String updateBio            = "Used to live in a pineapple under the sea!";
+
+
+    @Rule
+    public ActivityTestRule<MainActivity> testRule = new ActivityTestRule<>(MainActivity.class);
+
+    /////////////////////////////////////////// Tests //////////////////////////////////////////////
+
+    @Test
+    public void editProfileWoutPicture() {
+        sleep(3500);        // sleep on startup and between changing activities
```

Comment:
are this time sleep a must? We are running now the tests before every build of the app.. we should try to avoid using sleep.

Reply:
Good point.
I solved this issue, please see next commit.